### PR TITLE
PIM-5715: Enforce coupling rules on PIM Catalog component - Part II - Easy picks

### PR DESCRIPTION
**Pull request by damien-carcel**
**Description (for Contributor and Core Developer)**

To extract the domain layer and ensure its consistency, some coupling rules have been defined here https://akeneo.atlassian.net/wiki/pages/viewpage.action?pageId=76382220.
To check if the rules are well enforced we use the coupling detector https://github.com/akeneo/php-coupling-detector. This file 
https://github.com/akeneo/pim-community-dev/blob/master/.php_cd_transitional.php lists the authorized dependencies per components.

To check if the rules are enforced, use the following command:
```bash
bin/php-coupling-detector detect src/Pim/Component/Catalog --config-file=.php_cd_transitional.php
```

This PR intends to low the coupling issues (**24** violations before, **20** in this PR).

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Yes
| Added Behats                      | No
| Changelog updated                 | Yes
| Review and 2 GTM                  | Yes
| Micro Demo to the PO (Story only) | N/A
| Migration script                  | Yes
| Tech Doc                          | No

### CODE REVIEW

File: src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Akeneo\Component\Localization\Presenter;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class TranslatablePresenter
```
Comment (skeleton) [inline src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php]: Add a better description please

Comment (damien-carcel) [inline src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php]: On my way, sorry.

File: src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        $parameters = isset($options['parameters']) ? $options['parameters'] : [];
+        $domain     = isset($options['domain']) ? $options['domain'] : [];
+
+        return $this->translator->trans($value, $parameters, $domain);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($attributeType)
```
Comment (skeleton) [inline src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php]: this service is not tagged, this method is useless

File: src/Pim/Bundle/CatalogBundle/Resources/config/localization/presenters.yml [hunk 3]
```diff
@@ -66,3 +68,9 @@ services:
             - ['date_min', 'date_max']
         tags:
             - { name: pim_catalog.localization.presenter, type: 'attribute_option' }
+
+    pim_catalog.localization.presenter.translatable:
+        class: %pim_catalog.localization.presenter.translatable.class%
+        arguments:
+            - '@translator'
+            - ['pim_catalog_metric']
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Resources/config/localization/presenters.yml]: I don't understand that

Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Resources/config/localization/presenters.yml]: Useless in fact. I will remove it.

File: src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php (hunk 2 above)
Comment (damien-carcel) [inline src/Akeneo/Component/Localization/Presenter/TranslatablePresenter.php]: But it implements the interface, so I don't really have a choice. There is already the same problem with the DatePresenter.

File: src/Akeneo/Component/Localization/TranslatorProxy.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * Symfony Translator proxy for the Localization component.
+ *
+ * @author    Damien Carcel <damien.carcel@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class TranslatorProxy
+{
+    /** @var string[] */
+    protected $attributeTypes;
```
Comment (jjanvier) [inline src/Akeneo/Component/Localization/TranslatorProxy.php]: to remove, unused

File: src/Pim/Component/Catalog/Localization/Presenter/MetricPresenter.php [hunk 5]
```diff
@@ -15,19 +15,22 @@
...
+     * @param NumberFactory   $numberFactory
+     * @param array           $attributeTypes
+     * @param TranslatorProxy $translatorProxy
      */
-    public function __construct(NumberFactory $numberFactory, array $attributeTypes, TranslatorInterface $translator)
-    {
+    public function __construct(
+        NumberFactory $numberFactory,
+        array $attributeTypes,
+        TranslatorProxy $translatorProxy
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/Localization/Presenter/MetricPresenter.php]: you should switch the order of $attributeTypes and TranslatorProxy ;)

**Comment by jjanvier** [comment]
GTM ! Very good work :)

File: src/Pim/Component/Catalog/Localization/Presenter/MetricPresenter.php (hunk 5 above)
Comment (damien-carcel) [inline src/Pim/Component/Catalog/Localization/Presenter/MetricPresenter.php]: It is an added argument that the parent doesn't have, that's why it is at the end :).

**Comment by skeleton** [comment]
GTM :)

**Pull request merged**